### PR TITLE
Add type definition tests

### PR DESCRIPTION
**Pull request by SamVerschueren**
This PR adds [tsd-check](https://github.com/SamVerschueren/tsd-check) which tests the type definitions.

### CODE REVIEW

**Comment by SamVerschueren** [comment]
I fixed the failing tests, but GitHub looks to be slow in picking up the changes.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
 		"node": ">=6"
 	},
 	"scripts": {
-		"test": "xo && ava"
+		"test": "xo && tsd-check && ava"
```
Comment (SamVerschueren) [inline package.json]: Wasn't sure about the order of execution.

Comment (sindresorhus) [reply package.json]: I would put it last. I don't want failing types to prevent me from running tests when developing.

Comment (SamVerschueren) [reply package.json]: Ah, good point! Makes sense. Fixed it.

**Pull request merged**